packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -123,19 +184,81 @@ class DatePickerInput extends React.Component {
     }
   }
 
-  handleOnChange(event) {
-    if (!DateUtil.validDateInput(event.target.value)) {
-      return;
+  handleOnKeyDown(event) {
+    if (this.props.onKeyDown) {
+      this.props.onKeyDown(event);
     }
+  }
+
+  handleDayChange(event) {
+    const inputValue = event.target.value.replace(/\D/gm, '');
+    this.handleDateChange(event, DateUtil.inputType.DAY, inputValue);
+  }
 
-    if (this.props.onChange) {
-      this.props.onChange(event);
+  handleMonthChange(event) {
+    const inputValue = event.target.value.replace(/\D/gm, '');
+    this.handleDateChange(event, DateUtil.inputType.MONTH, inputValue);
+  }
+
+  handleYearChange(event) {
+    const inputValue = event.target.value.replace(/\D/gm, '');
+    this.handleDateChange(event, DateUtil.inputType.YEAR, inputValue);
+  }
+
+  handleOnFocus(event) {
+    this.setState({ isFocused: true });
+    if (this.props.onFocus) {
+      this.props.onFocus(event);
     }
   }
 
-  handleOnKeyDown(event) {
-    if (this.props.onKeyDown) {
-      this.props.onKeyDown(event);
+  handleOnBlur(event) {
+    this.setState({ isFocused: false });
+    if (this.props.onBlur) {
+      this.props.onBlur(event);
+    }
+  }
+
+  handleDateChange(event, type, value) {
+    if (type === DateUtil.inputType.DAY) {
+      this.setState({
+        day: value,
+      }, this.moveFocusOnChange(value, type, this.variant));
+    } else if (type === DateUtil.inputType.MONTH) {
+      this.setState({
+        month: value,
+      }, this.moveFocusOnChange(value, type, this.variant));
+    } else if (type === DateUtil.inputType.YEAR) {
+      this.setState({
+        year: value,
+      }, this.moveFocusOnChange(value, type, this.variant));
+    }
+
+    const day = type === DateUtil.inputType.DAY ? value : this.state.day;
+    const month = type === DateUtil.inputType.MONTH ? value : this.state.month;
+    const year = type === DateUtil.inputType.YEAR ? value : this.state.year;
+    const dateFormat = DateUtil.getFormatByLocale(this.props.intl.locale);
+
+    let tempDate;
+    let formattedDate;
+
+    if (this.variant === 'YYYY-MM-DD') {
+      if (month && day && year && day.length === 2) {
+        tempDate = DateUtil.formatISODate(year.concat('-', month).concat('-', day), 'YYYY-MM-DD');
```

Comment:
Logic in Lines 247 and 248 seem redundant with Lines 251 and 252. Can these conditionals be made simpler then?

Reply:
I tried to clean this up a little bit! [89c3693](https://github.com/cerner/terra-framework/pull/1004/commits/89c3693f31162adfc53f90f681cdfda21588b6b0)